### PR TITLE
Frontend can retrieve regional information about a member of staff

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

* Adds new middleware `populateUserRegion`, which populates the session with the user's probation region
* Updates `extractCallConfig` and `RestClient` to use this information in the session to send a new header `X-USER-REGION` with each API call

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by bryangaledxw** [comment]
[Frontend can retrieve regional information about a member of staff](https://trello.com/c/mLvqd7ez/715-frontend-can-retrieve-regional-information-about-a-member-of-staff)

File: server/testutils/factories/user.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Factory } from 'fishery'
+import { faker } from '@faker-js/faker/locale/en_GB'
+import type { User } from '@approved-premises/api'
+import referenceData from './referenceData'
+
+export default Factory.define<User>(() => ({
```
Comment (richpjames) [inline server/testutils/factories/user.ts:6]: Would a type of `TemporaryAccommodationUser` be more fitting here?

File: integration_tests/tests/temporary-accommodation/manage/report.cy.ts [hunk 2]
```diff
@@ -5,12 +5,14 @@ import Page from '../../../../cypress_shared/pages/page'
...
 import referenceData from '../../../../server/testutils/factories/referenceData'
 import DashboardPage from '../../../../cypress_shared/pages/temporary-accommodation/dashboardPage'
+import stubActingUser from '../../../../cypress_shared/utils/stubActingUser'
 
 context('Report', () => {
   beforeEach(() => {
     cy.task('reset')
     cy.task('stubSignIn')
     cy.task('stubAuthUser')
+    stubActingUser()
```
Comment (richpjames) [inline integration_tests/tests/temporary-accommodation/manage/report.cy.ts:15]: Seeing all these identical changes in different makes wonder if we should have single 'setup' task that does these 4 tasks... maybe something for later!

File: server/utils/restUtils.test.ts [hunk 3]
```diff
@@ -1,13 +1,26 @@
 import { createMock } from '@golevelup/ts-jest'
 import { Request } from 'express'
+import probationRegionFactory from '../testutils/factories/probationRegion'
 import extractCallConfig from './restUtils'
 
 describe('extractCallConfig', () => {
-  it('extracts the token from the given request', () => {
+  it('extracts the token and from the given request', () => {
```
Comment (richpjames) [inline server/utils/restUtils.test.ts]: 'and the probation region'

**Review by richpjames — APPROVED** [review]
LGTM, have left a couple of minor comments

File: integration_tests/tests/temporary-accommodation/manage/report.cy.ts (hunk 2 above)
Comment (bryangaledxw) [reply integration_tests/tests/temporary-accommodation/manage/report.cy.ts:15]: Ah, yeah. Just had a go at integrating that with this PR, turned into a bit of a rabbit hole - will get this in and have another go

**Pull request merged**